[rapidcheck] Add initial support for RapidCheck

Functorius opened the pull request:
Create a portfile for the C++ property-based testing library RapidCheck. This port is relatively straight forward but there are a couple of caveats:
 * The current port uses a branch from a fork that I modified as the CMake installation logic had some problems. The library was almost exclusively used by being added as a subfolder so the CMake export machinery was probably not well tested.
 * The changes are going to get merged into the main repository eventually and then it would be possible to change where the source code is downloaded.

I am unaware when the main RapidCheck library maintainer will accept my pull request, so it is better to have this library in VCPKG sooner, rather than later.

This is related to #4595.

Commit:
Add initial support for RapidCheck

Create a portfile for the C++ property-based testing library RapidCheck.
This port is relatively straight forward but there are a couple of caveats:
 * The current port uses a branch from a fork that I modified as the CMake installation logic had some problems.
   The library was almost exclusively used by being added as a subfolder so the CMake export machinery was probably not well tested.
 * The changes are going to get merged into the main repository eventually and then it would be possible to change where the source
   code is downloaded.

I am unaware when the main RapidCheck library maintainer will accept my pull request, so it is better to have this library in VCPKG
sooner, rather than later.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=4605) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=4605) before we can accept your contribution.<br/><br/>:x: Thorius [sign now](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=4605)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/vcpkg?pullRequest=4605) it.</sub>

Functorius committed:
Update the RapidCheck portfile to download the library sources for the main repository

 * Since the maintainer of RapidCheck very quickly merged the required CMake modifications
   to enable the proper installation of the library, it is better to now use the official
   repository's master branch.
 * Change the versioning of the library to be based on the date of the git reference chosen
   since rapid check does not have an explicit versioning scheme.

Thorius commented:
Just to mention it here, but I will probably make 1 extra commit here since the RapidCheck project's CMake files will probably change slightly in about a 1 day. So I would probably prefer to make the changes now instead of making a separate commit later on.

Commit:
Update rapidcheck and modify configuration step

Thorius commented:
@ras0219-msft , I pushed the updated portfile, taking into account recent changes of RapidCheck's CMake files. Everything should be all set for merging now.

ras0219-msft committed:
Merge branch 'master' of https://github.com/Microsoft/vcpkg into dev/roschuma/4605

ras0219-msft committed:
[rapidcheck] Tidy

ras0219-msft commented:
Thanks for the PR!
